src/pages/data-management/index.jsx

```diff
@@ -173,11 +155,19 @@ const DataManagementPage = ({ route }) => {
         route={route}
         title='Data Management'
       />
-      <LoadingModal
-        visible={Boolean(projectSaving || sampleSaving)}
-        message={projectSaving ?? sampleSaving ?? ''}
-      />
-
+      {projectSaving || sampleSaving ? (
+        <center>
+          <Space direction='vertical'>
+            <ClipLoader
+              size={50}
+              color='#8f0b10'
+            />
+            <Text>
+              Loading...
+            </Text>
+          </Space>
+        </center>
+      ) : (<></>)}
       <NewProjectModal
         visible={newProjectModalVisible}
         firstTimeFlow={projectsList.ids.length === 0}
```

Comment:
It seems that `onCreate` and `onCancel` on this component is doing the same thing. Rather than having a separate `createNewProject` function, it'll be good to follow the pattern in `onCancel`

Reply:
yup definitely, I fixed this along with other small things I noticed that were left to fix